universal-application-tool-0.0.1/app/views/admin/programs/ProgramApplicationListView.java

```diff
@@ -0,0 +1,116 @@
+import j2html.tags.Tag;
+import java.util.NoSuchElementException;
+import models.Application;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import play.twirl.api.Content;
+import services.WellKnownPaths;
+import views.BaseHtmlView;
+import views.admin.AdminLayout;
+import views.components.LinkElement;
+import views.style.Styles;
+
+public final class ProgramApplicationListView extends BaseHtmlView {
+  private final AdminLayout layout;
+  private final Logger log = LoggerFactory.getLogger(ProgramApplicationListView.class);
+
+  @Inject
+  public ProgramApplicationListView(AdminLayout layout) {
+    this.layout = layout;
+  }
+
+  public Content render(long programId, ImmutableList<Application> applications) {
+    Tag contentDiv =
+        div()
+            .withClasses(Styles.PX_20)
+            .with(
+                h1("All Applications").withClasses(Styles.MY_4),
+                each(applications, this::renderApplicationListItem),
+                renderDownloadButton(programId));
+
+    return layout.render(head(layout.tailwindStyles()), body(contentDiv));
+  }
+
+  public Tag renderDownloadButton(long programId) {
+    String link = controllers.admin.routes.AdminApplicationController.downloadAll(programId).url();
+    return new LinkElement()
+        .setId("download-all-button")
+        .setHref(link)
+        .setText("Download all (CSV)")
+        .asButton();
+  }
+
+  public Tag renderApplicationListItem(Application application) {
+    String downloadLinkText = "Download (PDF)";
+    long applicationId = application.id;
+    String applicantName;
+    try {
+      String firstName =
+          application.getApplicantData().readString(WellKnownPaths.APPLICANT_FIRST_NAME).get();
+      String lastName =
+          application.getApplicantData().readString(WellKnownPaths.APPLICANT_LAST_NAME).get();
+      applicantName = String.format("%s, %s", lastName, firstName);
+    } catch (NoSuchElementException e) {
+      log.error("Application {} does not include an applicant name.");
+      applicantName = "<Anonymous Applicant>";
+    }
+    String lastEditText;
+    try {
+      lastEditText =
+          application
+              .getApplicantData()
+              .readString(WellKnownPaths.APPLICATION_SUBMITTED_TIME)
+              .get();
+    } catch (NoSuchElementException e) {
+      log.error("Application {} submitted without submission time marked.", applicationId);
+      lastEditText = "<ERROR>";
+    }
+    lastEditText = "Last edited " + lastEditText;
+
+    Tag topContent =
+        div(
+                div(
+                    div(applicantName)
+                        .withClasses(
+                            Styles.TEXT_BLACK, Styles.FONT_BOLD, Styles.TEXT_XL, Styles.MB_2)),
+                p().withClasses(Styles.FLEX_GROW))
+            .withClasses(Styles.FLEX);
+
+    Tag bottomContent =
+        div(
+                p(lastEditText).withClasses(Styles.TEXT_GRAY_700, Styles.ITALIC),
```

Comment:
Nit. (Non-blocking) Can you add an Issue for someone to create a color palette?

Ideally we'd never directly use `TEXT_[COLOR_###]` or `BG_[COLOR_###]` but we'd have...

```
Theme.TEXT_PRIMARY = StyleUtils.joinStyles(Style.TEXT_BLACK, StyleUtils.dark(Styles.TEXT_GREY_100));
Theme.BG_PRIMARY = StyleUtils.joinStyles(Style.BG_WHITE, StyleUtils.dark(Styles.BG_GRAY_600));
```

I just haven't gotten around to it yet.

Reply:
#534, done.